start

```diff
@@ -1,14 +1,23 @@
 #!/bin/bash
 
+set -e
+
+# this replicates the default location logic from https://docs.gitea.io/en-us/config-cheat-sheet/
+# any setting can be overridden just by setting its variable before calling this script
+: ${GITEA_APP_PATH:=../gitea/gitea}
```

Comment:
This is awkward!

In our deployments, this will be `/srv/gitea/gitea`. On my dev machine I have ~/src/gitea/ and ~/src/bids-hook.

I don't know. Maybe this should be a mandatory input? But then again, if you on a server where you're explicitly setting GITEA_CUSTOM for gitea, you can also set it for bids-hook and then GITEA_APP_PATH is irrelevant.

Reply:
I realize now that I didn't mention this before, but: the `start` script is meant to disappear from the repo at some point! It's just scaffolding for manual, local testing on my own machine (or yours!) while we develop a v1. Hence all the hardcoded values and dummy secret and token.

So actual deployment situations are out of scope for this file.

Reply:
That makes sense. In that case, the part that mirrors Gitea's defaults logic should probably make its way into Go instead of shell. I could probably copy it straight out of their codebase.

And the environment variables that remain would migrate to a systemd .service file? Is it weird to put credentials in a .service file?

Reply:
It looks like systemd has a whole system in place for credentials. According to [System and Service Credentials](https://systemd.io/CREDENTIALS/):

> Within unit files, there are four settings to configure service credentials.
> 1. `LoadCredential=` may be used to load a credential from disk, from an `AF_UNIX` socket, or propagate them from a system credential.
> 2. `SetCredential=` may be used to set a credential to a literal string encoded in the unit file. **Because unit files are world-readable (both on disk and via D-Bus), this should only be used for credentials that aren’t sensitive, i.e. public keys/certificates – but not private keys.**
> 3. `LoadCredentialEncrypted=` is similar to `LoadCredential=` but will load an encrypted credential, and decrypt it before passing it to the service. For details on credential encryption, see below.
> 4. `SetCredentialEncrypted=` is similar to `SetCredential=` but expects an encrypted credential to be specified literally. Unlike `SetCredential=` it is thus safe to be used even for sensitive information, because even though unit files are world readable, the ciphertext included in them cannot be decoded unless access to TPM2/encryption key is available.

So, systemd wants to make credentials available as files, and currently bids-hook wants to accept credentials as environment variables. I guess I can change the environment variable to contain a _path_ to the secret, rather than the secret itself, and that should be sufficiently flexible for manual testing and also for systemd-controlled execution.